### PR TITLE
Make use of standard RecyclerView Holder optimizations.

### PR DESCRIPTION
**Pull request by CodingDoug**
An important optimization that RecyclerView offers is the ability to cache the results of (potentially expensive) calls to findViewById() inside a ViewHolder so that they don't have to be repeated every time data is bound to a recycled view.

### CODE REVIEW

File: app/src/main/java/com/firebase/uidemo/database/ChatActivity.java [hunk 1]
```diff
@@ -226,49 +226,50 @@ public String getText() {
...
-            int color;
+        public void setIsSender(boolean isSender) {
+            final int color;
             if (isSender) {
-                color = ContextCompat.getColor(mView.getContext(), R.color.material_green_300);
-
-                left_arrow.setVisibility(View.GONE);
-                right_arrow.setVisibility(View.VISIBLE);
-                messageContainer.setGravity(Gravity.END);
+                color = ContextCompat.getColor(itemView.getContext(), R.color.material_green_300);
```
Comment (samtstern) [inline app/src/main/java/com/firebase/uidemo/database/ChatActivity.java]: While we're feeling cache-y, might as well cache these two colors on ViewHolder creation (or even at a higher level)

**Comment by samtstern** [comment]
LGTM, made one comment for an additional improvement while I have you here.

File: app/src/main/java/com/firebase/uidemo/database/ChatActivity.java (hunk 1 above)
Comment (CodingDoug) [inline app/src/main/java/com/firebase/uidemo/database/ChatActivity.java]: Considered that, but those could actually theoretically change when the configuration changes as well.  If we want to assume they don't, then they're cachable as well.

Comment (samtstern) [inline app/src/main/java/com/firebase/uidemo/database/ChatActivity.java]: Since this is a very simple sample app let's assume they don't.

**Pull request merged**